File: Connect.java
class Solution{
    public Node connect(TreeNode root){
        if(root == null) return;
        Node level  = root;
        while(level.left != null){
            Node curr  = level;
            while(curr != null){
                curr.left.next = curr.right;
                if(curr.next != null){
                    curr.right.next = curr.next.left;
                }
                curr = curr.next;
            }
            level = level.left;
        }
        return root;
    }
}

File: connect.swift
// Time Complexity : 0(n)
// Space Complexity : o(1)
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach

       func connect(_ root: Node?) -> Node? {
        if root == nil {
            return root
        }
        var level = root
        var current = root
        var previous : Node? = nil
        while level != nil {
            while current != nil {
                current?.left?.next = current?.right
                previous?.right?.next = current?.left
                previous = current
                current = current?.next
            }
            level = level?.left
            current = level
            previous = nil
        }
        return root
    }
  
   
